File: Python/day_6.py
import string

def solve():

    with open("../inputs/day_6.txt", "r") as f:
        line = f.readline().strip()

    prev = {char: -1 for char in string.ascii_lowercase}
    count = 0
    for i, char in enumerate(line):
        if prev[char] == -1 or prev[char] < i - count:
            prev[char] = i
            count += 1
            if count == 14:
                print(f"Your result is: {i + 1}")
                break
        else:
            count = i - prev[char]
            prev[char] = i


if __name__ == "__main__":
    solve()

File: Rust/src/day_6.rs
use itertools::Itertools;

pub fn solution() {

    let n = 4;
    let answer = include_str!("../../inputs/day_6.txt")
        .trim()
        .chars()
        .collect::<Vec<char>>()
        .windows(n)
        .enumerate()
        .filter(|(_, window)| window.into_iter().unique().count() == n)
        .map(|(i, _)| i + n)
        .next().unwrap();

    println!("{:?}", answer);

}